### PR TITLE
wip: Establish Security section for OTel documentation

### PR DESCRIPTION
**Pull request by mjingle**
OpenTelemetry (OTel) currently does not have public documentation on security recommendations. A markdown file in the OpenTelemetry Collector repository (linked below) does address some concerns, but is not easily findable by end users.

The goal of this PR is to establish hosting and configuration security recommendations for OTel Collector end users within the OpenTelemetry public documentation site. While the inspiration document contains both end-user and Component Developer security recommendations, this public documentation adapts the existing content focused on end users and links back to the inspiration document for Component Developers.

**NOTE**: The inspiration document also does not cover topics in detail and I need help filling in those blanks. Refer to any `<!-- TODO: -->` within the files that indicates knowledge gaps. From there, the page content can be re-organized, given the available information.

This PR adds:

- a Security Overview page
- an OpenTelemetry Collector hosting best practices page
- an OpenTelemetry Collector configuration best practices page

This proposed structure tries to be flexible for future Security content pages to be added. (Additional content doesn't have to focus on OTel Collector. Other Security-related areas in OTel can be addressed also, but that's not within the scope of this PR.)

## Identified Areas of Potential Help

I already know this PR needs some extra help and this is my first time contributing a PR to this project. These include:

- Filling in content gaps. (Refer to **NOTE:** above.)
- Correct wiring for the menu. My goal is to have Security as a top level menu option.
- Links to related content within the OTel community documentation, if applicable.
- Amending the [inspiration document](https://github.com/open-telemetry/opentelemetry-collector/blob/main/docs/security-best-practices.md) in the OTel Collector repo to focus on Component Developers only.

## Source Materials

https://github.com/open-telemetry/opentelemetry-collector/blob/main/docs/security-best-practices.md

## Associated Issues

Initial issue: https://github.com/open-telemetry/opentelemetry.io/issues/3479
Related issue: https://github.com/open-telemetry/opentelemetry.io/issues/3227

---

**Preview**: https://deploy-preview-3652--opentelemetry.netlify.app/docs/security/

### CODE REVIEW

File: content/en/docs/security/otel-collector-configuration-best-practices.md [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+### Collector's Internal Telemetry
+
+<!--- INSERT RECOMMENDATIONS HERE. For example:
+
+1. Remove zPages.
+1. Remove configuration endpoints.
+-->
+
+### Forwarding [component developers?]
```
Comment (mjingle) [inline content/en/docs/security/otel-collector-configuration-best-practices.md:124]: Am not familiar with Forwarding. Is this content related to component developers? If so, I can remove it from this PR for end users.

File: content/en/docs/security/otel-collector-configuration-best-practices.md [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+
+### Safeguarding around resource utilization
+
+Processors also offer safeguards around resource utilization. After implementing
+[safeguards around resource utilization in your hosting](/security/otel-collector-hosting-best-practices/),
+ensure your OpenTelemetry Collector configuration uses these safeguards.
+
+<!-- start same page content in otel-collector-hosting-best-practices -->
+
+The `batch` and especially `memory_limiter` processor help ensure that the
```
Comment (jpkrohling) [inline content/en/docs/security/otel-collector-configuration-best-practices.md:89]: We probably don't want to mention the memory limiter anymore, as we are in talks to deprecate/remove it.

cc @dmitryax

File: content/en/docs/security/otel-collector-configuration-best-practices.md [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+### Health and Telemetry
+
+The initial extensions provided health check information, Collector metrics and
+traces, and the ability to generate and collect profiling data. When enabled
+with their default settings, all of these extensions except the health check
+extension are only accessibly locally to the OpenTelemetry Collector. Care
+should be taken when configuring these extensions for remote access as sensitive
+information may be exposed as a result.
+
+### Collector's Internal Telemetry
```
Comment (jpkrohling) [inline content/en/docs/security/otel-collector-configuration-best-practices.md:116]: @codeboten, you mentioned earlier this week that we can export traces from the Collector. Do you have docs that we could use here?

File: content/en/docs/security/otel-collector-configuration-best-practices.md (hunk 1 above)
Comment (jpkrohling) [reply content/en/docs/security/otel-collector-configuration-best-practices.md:124]: I would keep this out, at least for now.

File: content/en/docs/security/otel-collector-configuration-best-practices.md [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+
+### Forwarding [component developers?]
+
+A forwarding extension is typically used when some telemetry data not natively
+supported by the OpenTelemetry Collector needs to be collected. For example, the
+`http_forwarder` extension can receive and forward HTTP payloads. Forwarding
+extensions are similar to receivers and exporters so the same security
+considerations apply.
+
+### Observers [component developers?]
```
[the remark this answers is not in the document]
Comment (jpkrohling) [reply content/en/docs/security/otel-collector-configuration-best-practices.md:132]: This is for end-users. Perhaps an example would be useful here, for users to understand what they can do with observers.

Here's one such example, where a redis receiver is created if a redis pod is found on a Kubernetes cluster:

https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/extension/observer/k8sobserver

cc @rmfitzpatrick , @dmitryax

File: content/en/docs/security/otel-collector-configuration-best-practices.md [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+
+An observer is capable of performing service discovery of endpoints. Other
+components of the OpenTelemetry Collector such as receivers MAY subscribe to
+these extensions to be notified of endpoints coming or going. Observers MAY
+require certain permissions in order to perform service discovery. For example,
+the `k8s_observer` requires certain RBAC permissions in Kubernetes, while the
+`host_observer` requires the OpenTelemetry Collector to run in privileged mode.
+
+### Subprocesses [component developers?]
```
[the remark this answers is not in the document]
Comment (jpkrohling) [reply content/en/docs/security/otel-collector-configuration-best-practices.md:141]: This can be removed. I don't think we ship any such components anymore.

File: content/en/docs/security/otel-collector-hosting-best-practices.md [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+following practices to better secure your hosting instance.
+
+## Storing Configuration Information Securely
+
+<!--- TODO: SHOULD ensure sensitive configuration information is stored securely. How? -->
+
+## Permissions
+
+<!--- TODO: SHOULD not run the OpenTelemetry Collector as root/admin user. Why? (Give the reader motivation.) How do you do that?
+- NOTE: MAY require privileged access for some components -->
```
Comment (jpkrohling) [inline content/en/docs/security/otel-collector-hosting-best-practices.md:17]: The Collector SHOULD NOT require privileged access, except where the data it's obtaining is in a privileged location. For instance, in order to get pod logs by mounting a node volume, the Collector daemonset needs enough privileges to get that data.

The rule of least privilege applies here.

File: content/en/docs/security/otel-collector-hosting-best-practices.md [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+
+## Receivers and Exporters
+
+<!--- TODO: SHOULD limit exposure of servers to authorized users. How do you do that? -->
```
Comment (jpkrohling) [inline content/en/docs/security/otel-collector-hosting-best-practices.md:21]: - Authentication (bearer token auth extension, basic auth extension)
- By restricting the IPs the collector is running

File: content/en/docs/security/_index.md [hunk 8]
```diff
@@ -0,0 +1,21 @@
+---
+title: Security
+description: How to use OpenTelemetry Collector securely.
+weight: 90
```
Comment (chalin) [inline content/en/docs/security/_index.md:4]: Let's try to place this right after the Collector section:

```suggestion
weight: 10.1
```

File: content/en/docs/security/_index.md [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+---
+
+## Security
```
Comment (chalin) [inline content/en/docs/security/_index.md:7]: The page title is enough, no need to repeat the title in a section heading:

```suggestion
```

File: content/en/docs/security/otel-collector-configuration-best-practices.md [hunk 10]
```diff
@@ -0,0 +1,147 @@
...
+title: OpenTelemetry Collector Configuration Best Practices
+description:
+  Follow best practices to securely configure OpenTelemetry Collector.
+weight: 20
+---
+
+When setting up configuration for OpenTelemetry (OTel) Collector, consider the
+following practices to better secure your OTel Collector instance.
+
+## Configuration
```
Comment (chalin) [inline content/en/docs/security/otel-collector-configuration-best-practices.md:11]:
```suggestion
```

File: content/en/docs/security/otel-collector-hosting-best-practices.md [hunk 11]
```diff
@@ -0,0 +1,56 @@
+---
+title: OpenTelemetry Collector Hosting Best Practices
```
Comment (chalin) [inline content/en/docs/security/otel-collector-hosting-best-practices.md:2]:
```suggestion
title: OpenTelemetry Collector Hosting Best Practices
linkTitle: Collector hosting
```

File: content/en/docs/security/otel-collector-configuration-best-practices.md [hunk 12]
```diff
@@ -0,0 +1,147 @@
+---
+title: OpenTelemetry Collector Configuration Best Practices
```
Comment (chalin) [inline content/en/docs/security/otel-collector-configuration-best-practices.md:2]:
```suggestion
title: OpenTelemetry Collector Configuration Best Practices
linkTitle: Collector config
```

File: content/en/docs/security/otel-collector-configuration-best-practices.md (hunk 2 above)
Comment (mjingle) [reply content/en/docs/security/otel-collector-configuration-best-practices.md:88]: Will remove the `memory_limiter` mention then. Thank you!

File: content/en/docs/security/otel-collector-configuration-best-practices.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (mjingle) [reply content/en/docs/security/otel-collector-configuration-best-practices.md:131]: Amended the header here, since Observers are for end-users. 👍 

An example here would be good to have. Still need to incorporate what you shared, @jpkrohling, but I appreciate the resource you shared!

File: content/en/docs/security/otel-collector-configuration-best-practices.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (mjingle) [reply content/en/docs/security/otel-collector-configuration-best-practices.md:140]: Removed! Thank you, thank you!

File: content/en/docs/security/otel-collector-configuration-best-practices.md [hunk 13]
```diff
@@ -0,0 +1,147 @@
...
+  connection established SHOULD be over a secure and authenticated channel.
+- Unused receivers and exporters SHOULD be disabled to minimize the attack
+  vector of the Collector.
+- Receivers and Exporters MAY expose buffer, queue, payload, and/or worker
+  settings via configuration parameters. If these settings are available, users
+  should proceed with caution before modifying the default configuration values.
+  Improperly setting these values may expose the OpenTelemetry Collector to
+  additional attack vectors including resource exhaustion.
+
+It is possible that a receiver MAY require the OpenTelemetry Collector run in a
```
[the remark this answers is not in the document]
Comment (mjingle) [reply content/en/docs/security/otel-collector-configuration-best-practices.md:28]: Can we ask a specific person or team for confirmation about this assertion? (This sentence is located in the inspiration document here:https://github.com/open-telemetry/opentelemetry-collector/blob/main/docs/security-best-practices.md?plain=1#L124 )

File: content/en/docs/security/otel-collector-configuration-best-practices.md [hunk 14]
```diff
@@ -0,0 +1,147 @@
...
+TODO: SHOULD only enable the minimum required components. What are those?
+--->
+
+## Receivers and Exporters
+
+- Receivers and Exporters can be either push or pull-based. In either case, the
+  connection established SHOULD be over a secure and authenticated channel.
+- Unused receivers and exporters SHOULD be disabled to minimize the attack
+  vector of the Collector.
+- Receivers and Exporters MAY expose buffer, queue, payload, and/or worker
```
[the remark this answers is not in the document]
Comment (mjingle) [reply content/en/docs/security/otel-collector-configuration-best-practices.md:22]: Thank you, @jpkrohling! For values that are too wide, rather than:
> remember that values that are too wide (such as a large queue number) might mean that the Collector will attempt to use more memory than a setting with a smaller value.

Would it be more accurate to describe as: 
> remember that if the values are too wide, such as a large queue number, then the OTel Collector may use more memory than necessary, compared to using a setting with a smaller value.

**Comment by svrnm** [comment]
Thinking about the "where to put this" once again: We now have established `/docs/security/` and have the CVE there and will also have more content like https://github.com/open-telemetry/opentelemetry.io/pull/3675 as well. This PR contains security guidelines for the collector, so we have technically 2 places to house this:

1. `/docs/security/collector`
2. `/docs/collector/security`

I think we should go for option (2) but at the same time add a "Practices" page under /docs/security/practices that will list collector (and later other components as well) and link back to the component specific security, wdyt?

**Comment by jpkrohling** [comment]
Thinking out "loud", there are two moments I'm concerned about security for my collector:

* when evaluating the collector
* when hardening it right before going into production

For the first situation, I'd prefer the security documentation to be close to the getting started, so that I can have an idea whether the collector satisfies my security needs. During the second situation, I _think_ I'd prefer it close to the collector documentation again, as I probably don't care about the security aspects of the SDK or other parts.

In summary: I'm with you that placing it closer to the collector makes sense, as long as there's a reference to it from within the /security pages.

**Comment by cartersocha** [comment]
I vote put it in the collector docs and agree we can link out to it

**Comment by mjingle** [comment]
@svrnm My question is why not have the content in both locations? (This may not be possible with the functionality/architecture of the OTel docs site, but I still ask the question.) 

I agree that when setting up the OTel Collector, information on securing your OTel Collector is very relevant and should exist within the OTel Collector documentation.

In the original vision by establishing this Security section, we're also:

1) trying to highlight to potential OTel adopters that you can be secure and open source.
2) encouraging security best practices related to OpenTelemetry...beyond the OTel Collector.

**Comment by svrnm** [comment]
> @svrnm My question is why not have the content in both locations? (This may not be possible with the functionality/architecture of the OTel docs site, but I still ask the question.)

We can certainly do this (somehow), but this often comes with some additional questions: is this a redirect (so people go from /security to /collector), is it just the same content presented at 2 places, what is the point where this list outgrows that (I assume that we eventually will have security best practices across the docs). 

To make it short: yes, we can do that, but we might reconsider it eventually. 

> I agree that when setting up the OTel Collector, information on securing your OTel Collector is very relevant and should exist within the OTel Collector documentation.

> In the original vision by establishing this Security section, we're also:
> 
>     1. trying to highlight to potential OTel adopters that you can be secure and open source.
> 
>     2. encouraging security best practices related to OpenTelemetry...beyond the OTel Collector.

This seems to be untouched by the decision where this content lives. Even if we have it **only** in the collector section I would want to have a /security/practices page that gives a general overview and then links out to relevant pages like the collector security pages.

File: content/en/docs/security/_index.md [hunk 15]
```diff
@@ -1,4 +1,19 @@
 ---
 title: Security
-weight: 150
+description: How to use OpenTelemetry Collector securely.
```
Comment (theletterf) [inline content/en/docs/security/_index.md:3]:
```suggestion
description: How to use OpenTelemetry Collector securely.
```

Not being a fan of adverbs in titles aside, I think the section could use a more noun-based title. For example:

- OpenTelemetry security
- Security in OpenTelemetry
- Secure the OpenTelemetry Collector

Etc.

File: content/en/docs/security/_index.md [hunk 16]
```diff
@@ -1,4 +1,19 @@
...
+weight: 10.1
 ---
+
+When setting up the OpenTelemetry (OTel) Collector, consider implementing
+security best practices in both your hosting infrastructure and your OTel
+Collector configuration.
+
```
Comment (theletterf) [inline content/en/docs/security/_index.md:10]: Could we explain briefly in the intro what's at stake here? Why should someone care about securing the OpenTelemetry Collector? Is it a painful process? What knowledge is required?

File: content/en/docs/security/otel-collector-configuration-best-practices.md [hunk 17]
```diff
@@ -0,0 +1,139 @@
...
+  Follow best practices to securely configure OpenTelemetry Collector.
+weight: 20
+---
+
+When setting up configuration for OpenTelemetry (OTel) Collector, consider the
+following practices to better secure your OTel Collector instance.
+
+<!--
+TODO: SHOULD only enable the minimum required components. What are those?
+--->
```
Comment (theletterf) [inline content/en/docs/security/otel-collector-configuration-best-practices.md:14]: This is a GREAT question. I think the minimum is 1 receiver and 1 exporter?

File: content/en/docs/security/otel-collector-configuration-best-practices.md [hunk 18]
```diff
@@ -0,0 +1,139 @@
...
+
+## Receivers and Exporters
+
+- Receivers and Exporters can be either push or pull-based. In either case, the
+  connection established SHOULD be over a secure and authenticated channel.
+- Unused receivers and exporters SHOULD be disabled to minimize the attack
+  vector of the Collector.
```
Comment (theletterf) [inline content/en/docs/security/otel-collector-configuration-best-practices.md:21]:
```suggestion
- Receivers and exporters can be either push or pull-based. In either case, the
  connection established SHOULD be over a secure and authenticated channel.
- Unused receivers and exporters SHOULD be disabled to minimize the attack
  vector of the Collector.
```

I know specs capitalize verbs in RFC style, but I wouldn't do in docs. Could you lowercase all instances?

Comment (theletterf) [reply content/en/docs/security/otel-collector-configuration-best-practices.md:21]: Also, please write `exporters` with lower case 'e'.

File: content/en/docs/security/otel-collector-configuration-best-practices.md [hunk 19]
```diff
@@ -0,0 +1,139 @@
...
+- Receivers and Exporters MAY expose buffer, queue, payload, and/or worker
+  settings via configuration parameters. If these settings are available, users
+  should proceed with caution before modifying the default configuration values.
+  Improperly setting these values may expose the OpenTelemetry Collector to
+  additional attack vectors including resource exhaustion.
```
Comment (theletterf) [inline content/en/docs/security/otel-collector-configuration-best-practices.md:26]:
```suggestion
- Receivers and exporters may expose buffer, queue, payload, and worker
  settings using configuration parameters. If these settings are available, you
  should proceed with caution before modifying the default configuration values.
  Improperly setting these values may expose the OpenTelemetry Collector to
  additional attack vectors including resource exhaustion.
```

Best going with second person rather than addressing "users" if the guide is aimed at folks who can implement OTel security on their own.

File: content/en/docs/security/otel-collector-configuration-best-practices.md [hunk 20]
```diff
@@ -0,0 +1,139 @@
...
+
+It is possible that a receiver MAY require the OpenTelemetry Collector run in a
+privileged mode in order to operate, which could be a security concern, but
+today this is not the case.
```
Comment (theletterf) [inline content/en/docs/security/otel-collector-configuration-best-practices.md:30]:
```suggestion
A receiver might require that the OpenTelemetry Collector run in a
privileged mode, which could be a security concern, but
today this is not the case.
```

Why this is not the case today?

File: content/en/docs/security/otel-collector-configuration-best-practices.md [hunk 21]
```diff
@@ -0,0 +1,139 @@
...
+
+### Safeguards against denial of service attacks
+
+Users should bind receivers' servers to addresses that limit connections to
+authorized users, so that their Collectors aren't exposed to the public internet or to wider networks than necessary.
```
Comment (theletterf) [inline content/en/docs/security/otel-collector-configuration-best-practices.md:35]:
```suggestion
You should bind receivers' servers to addresses that limit connections to
authorized users, so that their Collectors aren't exposed to the public internet or to wider networks than necessary.
```

File: content/en/docs/security/otel-collector-configuration-best-practices.md [hunk 22]
```diff
@@ -0,0 +1,139 @@
...
+
+For example, if the OTLP receiver OTLP/gRPC server only has local clients, the
+`endpoint` setting SHOULD be bound to `localhost`:
```
Comment (theletterf) [inline content/en/docs/security/otel-collector-configuration-best-practices.md:38]:
```suggestion
For example, if the OTLP receiver OTLP/gRPC server only has local clients, bound
the `endpoint` setting to `localhost`:
```

In general, we want to avoid passive voice.  https://developers.google.com/style/voice

File: content/en/docs/security/otel-collector-configuration-best-practices.md [hunk 23]
```diff
@@ -0,0 +1,139 @@
...
+  otlp:
+    protocols:
+      grpc:
+        endpoint: localhost:4317
+```
+
+Generally, `localhost`-like addresses should be preferred over the `0.0.0.0`
+address. For more information, refer to the
+[CWE-1327: Binding to an Unrestricted IP Address](https://cwe.mitre.org/data/definitions/1327.html)
+definition.
```
Comment (theletterf) [inline content/en/docs/security/otel-collector-configuration-best-practices.md:51]:
```suggestion
Use `localhost` addresses instead of `0.0.0.0. For more
information, refer to the
[CWE-1327: Binding to an Unrestricted IP Address](https://cwe.mitre.org/data/definitions/1327.html)
definition.
```

File: content/en/docs/security/otel-collector-configuration-best-practices.md [hunk 24]
```diff
@@ -0,0 +1,139 @@
...
+
+### Encryption and Authentication
```
Comment (theletterf) [inline content/en/docs/security/otel-collector-configuration-best-practices.md:53]:
```suggestion
### Encryption and authentication
```

Please use sentence case for all headings.

File: content/en/docs/security/otel-collector-configuration-best-practices.md [hunk 25]
```diff
@@ -0,0 +1,139 @@
...
+  [OTel Collector Configuring Certificates](/collector/configuration/#setting-up-certificates)
+  documentation.
+
+- For authentication, use the OTel Collector's authentication mechanism, as
+  described in our
+  [OTel Collector Authentication](/collector/configuration/#authentication)
+  documentation.
+
+**NOTE**: A security risk may present if configuration parameters are modified
+improperly.
```
Comment (theletterf) [inline content/en/docs/security/otel-collector-configuration-best-practices.md:67]: I'd remove this unless we can be more specific about the risk.

File: content/en/docs/security/otel-collector-configuration-best-practices.md [hunk 26]
```diff
@@ -0,0 +1,139 @@
...
+
+## Processors
+
+Processors sit between receivers and exporters. They are responsible for
+processing the data in some way. From a security perspective, they are useful in
+a few ways.
+
+### Recommended Processors
+
+<!--- TODO: SHOULD configure recommended processors. If so, what are they? -->
```
Comment (theletterf) [inline content/en/docs/security/otel-collector-configuration-best-practices.md:77]: - redactionprocessor (never tried myself)
- transformprocessor (great for redacting PIIs and such)

File: content/en/docs/security/otel-collector-configuration-best-practices.md [hunk 27]
```diff
@@ -0,0 +1,139 @@
...
+
+### Scrubbing sensitive data
+
+It is common for an OpenTelemetry Collector to be used to scrub sensitive data
+before exporting it to a backend. This is especially important when sending the
+data to a third-party backend. The OpenTelemetry Collector **SHOULD** be
+configured to obfuscate or scrub sensitive data before exporting.
```
Comment (theletterf) [inline content/en/docs/security/otel-collector-configuration-best-practices.md:84]:
```suggestion
You can use the OpenTelemetry Collector to scrub sensitive data
before exporting it to a back end. Configure the Collector to
obfuscate or scrub sensitive data before exporting.
```

File: content/en/docs/security/otel-collector-configuration-best-practices.md [hunk 28]
```diff
@@ -0,0 +1,139 @@
...
+
+<!--- TODO: SHOULD configure obfuscation/scrubbing of sensitive metadata. How? Give more details and/or link to an existing document -->
+
+Use OpenTelemetry Collector's `redaction` processor to scrub sensitive data.
```
Comment (theletterf) [inline content/en/docs/security/otel-collector-configuration-best-practices.md:88]: I'd remove this line or provide an example in a subsection.

File: content/en/docs/security/otel-collector-configuration-best-practices.md [hunk 29]
```diff
@@ -0,0 +1,139 @@
...
+<!--- TODO: Extensions SHOULD NOT expose sensitive health or telemetry data. How? What can you do? -->
+
+### Health and Telemetry
+
+The initial extensions provided health check information, Collector metrics and
+traces, and the ability to generate and collect profiling data. When enabled
+with their default settings, all of these extensions except the health check
+extension are only accessibly locally to the OpenTelemetry Collector. Care
+should be taken when configuring these extensions for remote access as sensitive
+information may be exposed as a result.
```
Comment (theletterf) [inline content/en/docs/security/otel-collector-configuration-best-practices.md:122]:
```suggestion
extension are only accessibly locally to the OpenTelemetry Collector. Take
care to protect access to sensitive information when configuring  these
extensions, as they might expose it accidentally.
```

File: content/en/docs/security/otel-collector-configuration-best-practices.md [hunk 30]
```diff
@@ -0,0 +1,139 @@
...
+-->
+
+### Observers
+
+An observer is capable of performing service discovery of endpoints. Other
+components of the OpenTelemetry Collector such as receivers MAY subscribe to
+these extensions to be notified of endpoints coming or going. Observers MAY
+require certain permissions in order to perform service discovery. For example,
+the `k8s_observer` requires certain RBAC permissions in Kubernetes, while the
+`host_observer` requires the OpenTelemetry Collector to run in privileged mode.
```
Comment (theletterf) [inline content/en/docs/security/otel-collector-configuration-best-practices.md:139]:
```suggestion
An observer is a component that discovers services in endpoints. Other
components of the OpenTelemetry Collector such as receivers can subscribe to
these extensions to be notified of endpoints coming or going.

Observers might require certain permissions in order to discover services.
For example, the `k8s_observer` requires certain RBAC permissions in
Kubernetes, while the `host_observer` observer requires to run the
OpenTelemetry Collector in privileged mode.
```

File: content/en/docs/security/otel-collector-hosting-best-practices.md [hunk 31]
```diff
@@ -0,0 +1,61 @@
...
+linkTitle: Collector hosting
+description: Follow best practices to securely host OpenTelemetry Collector.
+weight: 10
+---
+
+When setting up hosting for OpenTelemetry (OTel) Collector, consider the
+following practices to better secure your hosting instance.
+
+## Storing Configuration Information Securely
```
Comment (theletterf) [inline content/en/docs/security/otel-collector-hosting-best-practices.md:11]: Reminder about sentence case.

**Review by theletterf — CHANGES_REQUESTED** [review]

File: content/en/docs/security/otel-collector-configuration-best-practices.md (hunk 18 above)
Comment (svrnm) [reply content/en/docs/security/otel-collector-configuration-best-practices.md:21]: Agreed, as long as we do not make a definition that "OpenTelemetry Collector Exporters" are a fixed term for us, let's have them lowercase

**Comment by theletterf** [comment]
@mjingle Thanks SO much for starting this. This is necessary. Hope you'll find my suggestions useful.

**Comment by svrnm** [comment]
@mjingle any updates for this PR?

**Comment by chalin** [comment]
All: I've rebased & resolved conflicts so that we can get a fresher view of this addition in the context of the updated website -- a lot has changed since this was submitted! :)

File: content/en/docs/security/otel-collector-configuration-best-practices.md (hunk 14 above)
[the remark this answers is not in the document]
Comment (mjingle) [reply content/en/docs/security/otel-collector-configuration-best-practices.md:22]: @jpkrohling Circling back to ask your feedback on the above ^

File: content/en/docs/security/otel-collector-configuration-best-practices.md (hunk 13 above)
[the remark this answers is not in the document]
Comment (mjingle) [reply content/en/docs/security/otel-collector-configuration-best-practices.md:28]: @jpkrohling Circling back to ask if there's an answer for the above-mentioned question?

File: content/en/docs/security/otel-collector-configuration-best-practices.md (hunk 3 above)
Comment (mjingle) [reply content/en/docs/security/otel-collector-configuration-best-practices.md:124]: @codeboten I'm all ears if you happen to have docs on this. :) Appreciate it in advance!

File: content/en/docs/security/_index.md (hunk 15 above)
Comment (mjingle) [reply content/en/docs/security/_index.md:3]: Incorporated. Thank you!

File: content/en/docs/security/_index.md (hunk 16 above)
Comment (mjingle) [reply content/en/docs/security/_index.md:10]: You ask great questions @theletterf! If you have the answers, I'm happy to put them in this document. I defer to the SMEs to answer these questions otherwise. At the very least, I will file issues if these questions aren't answered or incorporated by the time we merge this PR.

File: content/en/docs/security/otel-collector-configuration-best-practices.md (hunk 17 above)
Comment (mjingle) [reply content/en/docs/security/otel-collector-configuration-best-practices.md:14]: 👍 Included that as a statement in the ## Receivers and exporters section below.

File: content/en/docs/security/otel-collector-configuration-best-practices.md (hunk 18 above)
Comment (mjingle) [reply content/en/docs/security/otel-collector-configuration-best-practices.md:21]: Fixed. Thanks all!

File: content/en/docs/security/otel-collector-configuration-best-practices.md (hunk 19 above)
Comment (mjingle) [reply content/en/docs/security/otel-collector-configuration-best-practices.md:26]: Sounds good. Fixed. This is a good improvement for what I copied and pasted from the source document.

File: content/en/docs/security/otel-collector-configuration-best-practices.md (hunk 27 above)
Comment (mjingle) [reply content/en/docs/security/otel-collector-configuration-best-practices.md:84]: Incorporated! Thanks!

File: content/en/docs/security/otel-collector-hosting-best-practices.md (hunk 31 above)
Comment (mjingle) [reply content/en/docs/security/otel-collector-hosting-best-practices.md:11]: Appreciated! Fixed.

**Comment by theletterf** [comment]
@mjingle LMK if you need to discuss the draft or need more direct support — my comments are not meant to be blockers. Would be happy to commit to your fork if you want.

**Comment by cartermp** [comment]
Closing old PR (and some unfortunate conflicts, too)
